File: src/jray/ray/scenes/KugelTest.java
package jray.ray.scenes;

import java.util.ArrayList;
import java.util.List;

import jray.common.Object3D;
import jray.common.Sky;
import jray.common.Sphere;
import jray.common.Vect3;
import jray.ray.Camera;
import jray.ray.Scene;


public class KugelTest extends Scene {

	private List<Object3D> objects;
	private Camera cam;
	
	public KugelTest(){
		objects = new ArrayList<Object3D>();
		objects.add(new Sphere(new Vect3(0,0,0),1,0xFFFFFF00));
		cam = new Camera(new Vect3(0,0,-5),new Vect3(0,0,-3),new Vect3(0,1,0));
	}
	
	@Override
	public Camera getCamera() {
		return cam;
	}

	@Override
	public Sky getSky() {
		return null;
	}

	@Override
	public List<Object3D> getObjects() {
		return objects;
	}

}

File: JRayXLib/JRayXLib/Ray/Scenes/KugelTest.cs
using System.Collections.Generic;
using JRayXLib.Colors;
using JRayXLib.Scene;
using JRayXLib.Shapes;

namespace JRayXLib.Ray.Scenes
{
    public class KugelTest : OctreeScene
    {
        public KugelTest()
        {
            var objects = new List<I3DObject>
                {
                    //new Sphere(new Vect3(), 1, Color.Yellow, 0.5),
                    //new Plane(new Vect3{Y = -2}, new Vect3{Y = 1}, Color.Green),
                    new Cone(new Vect3{Y = 1}, new Vect3{X=-1, Y=-1, Z=-1}, 32, Color.Green),
                    new Sphere(new Vect3{X = 1, Z = 2}, 1, Color.Blue)
                };

            UpdateObjects(objects);
            Camera = new Camera(new Vect3 { Z = -10 }, new Vect3 { Z = -8 }, new Vect3 { Y = 1 });
        }
    }
}